processor/k8sprocessor/processor.go

```diff
@@ -245,12 +155,31 @@ func (kp *kubernetesprocessor) getAttributesForPodIP(ip string) map[string]strin
 	return pod.Attributes
 }
 
-func (kp *kubernetesprocessor) k8sIPFromAttributes(attrs pdata.AttributeMap) string {
-	ip := stringAttributeFromMap(attrs, k8sIPLabelName)
-	if ip == "" {
-		ip = stringAttributeFromMap(attrs, clientIPLabelName)
+type ipExtractor func(attrs pdata.AttributeMap) string
```

Comment:
nit: Generally we place type definitions at the top of the file. We could also place the type definition along with the two extractors in separate file. Up to you what you want to to if anything at all.

Reply:
Put those into a separate file (`ip_extractor.go`)